cpp/foxglove/include/foxglove/mcap.hpp

```diff
@@ -6,12 +6,22 @@
 #include <memory>
 #include <optional>
 #include <string>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
 
 #include "channel.hpp"
 
+/// @cond foxglove_internal
 enum foxglove_error : uint8_t;
 struct foxglove_mcap_writer;
 
+foxglove_error foxglove_mcap_write_metadata(
+  struct foxglove_mcap_writer* writer, const struct foxglove_string* FOXGLOVE_NONNULL name,
+  const struct foxglove_key_value* metadata, size_t metadata_len
```

Comment:
A lot of these C-like constructs in your forward declaration I believe are unnecessary in C++ (like prefixing the typename with `struct`). I made this change locally and it seemed to link OK.

```suggestion
  foxglove_mcap_writer* writer, const foxglove_string* name,
  const foxglove_key_value* metadata, size_t metadata_len
```